lib/spack/llnl/util/lock.py

```diff
@@ -167,12 +222,17 @@ def _poll_lock(self, op):
             return True
 
         except IOError as e:
-            if e.errno in (errno.EAGAIN, errno.EACCES):
-                # EAGAIN and EACCES == locked by another process
+            if e.errno == errno.EAGAIN:
+                # locked by another process, try again
                 pass
+            elif e.errno == errno.EACCES:
+                # permission denied
+                tty.warn("Unable to acquire the lock: {0}".format(str(e)))
```

Comment:
I'm not sure if the above two cases can be distinguished reliably.  Are you assuming that `EACCES` always means permission denied or is that documented somewhere?  The `fcntl` [docs](https://pubs.opengroup.org/onlinepubs/009695399/functions/fcntl.html) say:

> The fcntl() function shall fail if:
>
> [EACCES] or [EAGAIN]
> The cmd argument is F_SETLK; the type of lock ( l_type) is a shared (F_RDLCK) or exclusive (F_WRLCK) lock and the segment of a file to be locked is already exclusive-locked by another process, or the type is an exclusive lock and some portion of the segment of a file to be locked is already shared-locked or exclusive-locked by another process.

So it seems like *either* of these codes can result if the lock is held by another process (which is why they were initially together).  So I wonder if this won't generate a million warnings on some machines when we really don't want it to.

Reply:
IIRC I ran into the distinction early on when I was testing so I thought it might be helpful to log a warning if/when `EACCES` it is encountered.

Per http://man7.org/linux/man-pages/man3/errno.3.html:
- `EACCES` is "Permission denied (POSIX.1-2001) 
- `EAGAIN` is "Resource temporarily unavailable (may be the same value as `EWOULDBLOCK`) (POSIX.1-2001)

References include:
- https://docs.python.org/2/library/errno.html
- https://docs.python.org/3.8/library/errno.html
- http://man7.org/linux/man-pages/man3/errno.3.html
- https://www.gnu.org/software/libc/manual/html_node/Error-Codes.html